python3Packages.cryptography: 45.0.4 -> 46.0.1; adopt

mdaniels5757 opened the pull request:
Diff: https://github.com/pyca/cryptography/compare/refs/tags/45.0.4...refs/tags/46.0.1

Changelog: https://cryptography.io/en/latest/changelog/#v46-0-1

Closes: #446497

<s>Needs: #445749 (will not build without it)</s> Merged.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [X] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mdaniels5757 committed:
python3Packages.cryptography: 45.0.4 -> 46.0.1; adopt

Diff: https://github.com/pyca/cryptography/compare/refs/tags/45.0.4...refs/tags/46.0.1

Changelog: https://cryptography.io/en/latest/changelog/#v46-0-1

SuperSandro2000 commented on pkgs/development/python-modules/cryptography/default.nix:
```diff
@@ -98,6 +98,9 @@ buildPythonPackage rec {
     ];
-    maintainers = with maintainers; [ SuperSandro2000 ];
+    maintainers = with maintainers; [
+      SuperSandro2000
```
Please remove me from here

SuperSandro2000 commented on pkgs/development/python-modules/cryptography/vectors.nix:
```diff
@@ -33,6 +33,9 @@ buildPythonPackage rec {
     ];
-    maintainers = with maintainers; [ SuperSandro2000 ];
+    maintainers = with maintainers; [
+      SuperSandro2000
```
And here, too

mdaniels5757 committed:
python3Packages.cryptography: remove SuperSandro2000 as maintainer

Per their request: https://github.com/NixOS/nixpkgs/pull/446515#discussion_r2385813367

mweinelt commented:
CFFI is now merged.

vcunat commented:
CI says it has built on the two linux platforms.
